backend/editor/entries.py

```diff
@@ -82,59 +85,86 @@ async def create_node(self, label, entry, main_language_code):
         result = await get_current_transaction().run(" ".join(query), params)
         return (await result.data())[0]["n.id"]
 
-    async def parse_taxonomy(self, filename):
+    def parse_taxonomy(self, uploadfile=None):
         """
         Helper function to call the Open Food Facts Python Taxonomy Parser
         """
-        # Close current transaction to use the session variable in parser
-        await get_current_transaction().commit()
-
-        with SyncTransactionCtx() as session:
-            # Create parser object and pass current session to it
-            parser_object = parser.Parser(session)
-            try:
-                # Parse taxonomy with given file name and branch name
-                parser_object(filename, self.branch_name, self.taxonomy_name)
-                return True
-            except Exception as e:
-                raise TaxonomyParsingError() from e
+        if uploadfile is None:  # taxonomy is imported
+            base_url = (
+                "https://raw.githubusercontent.com/openfoodfacts/openfoodfacts-server"
+                "/main/taxonomies/"
+            )
+            filename = f"{self.taxonomy_name}.txt"
+            base_url += filename
+        else:  # taxonomy is uploaded
+            filename = uploadfile.filename
 
-    async def import_from_github(self, description):
-        """
-        Helper function to import a taxonomy from GitHub
-        """
-        base_url = (
-            "https://raw.githubusercontent.com/openfoodfacts/openfoodfacts-server"
-            "/main/taxonomies/"
-        )
-        filename = self.taxonomy_name + ".txt"
-        base_url += filename
         try:
             with tempfile.TemporaryDirectory(prefix="taxonomy-") as tmpdir:
                 # File to save the downloaded taxonomy
                 filepath = f"{tmpdir}/{filename}"
+                if uploadfile is None:
+                    # Downloads and creates taxonomy file in current working directory
+                    urllib.request.urlretrieve(base_url, filepath)
+                else:
+                    with open(filepath, "wb") as f:
+                        shutil.copyfileobj(uploadfile.file, f)
```

Comment:
I think this part should be isolated in a function to make code more readable.

The function should be a context manager so it handles the tempfile transparently

```python

    @contextlib.contextmanager
    def get_taxonomy_file(self, uploadfile=None):
        ...
        with tempfile.TemporaryDirectory(prefix="taxonomy-") as tmpdir:
           ...
           yield filepath

...

    def parse_taxonomy(self, uploadfile=None):
        with self.get_taxonomy_file(uploadfile) as filepath:
            ...
```